core/test/financial-templates/PrecisionErrors.js

```diff
@@ -0,0 +1,290 @@
+// Contracts to test
+const PricelessPositionManager = artifacts.require("PricelessPositionManager");
+
+// Other UMA related contracts and mocks
+const Store = artifacts.require("Store");
+const Finder = artifacts.require("Finder");
+const IdentifierWhitelist = artifacts.require("IdentifierWhitelist");
+const MarginToken = artifacts.require("ExpandedERC20");
+const SyntheticToken = artifacts.require("SyntheticToken");
+const TokenFactory = artifacts.require("TokenFactory");
+
+contract("Measuring ExpiringMultiParty precision loss", function(accounts) {
+  const { toWei, toBN, utf8ToHex } = web3.utils;
+  const contractDeployer = accounts[0];
+  const sponsor = accounts[1];
+  const other = accounts[3];
+  const collateralOwner = accounts[4];
+
+  // Contracts
+  let collateral;
+  let pricelessPositionManager;
+  let identifierWhitelist;
+
+  // Initial constant values
+  const syntheticName = "UMA test Token";
+  const syntheticSymbol = "UMATEST";
+  const withdrawalLiveness = 1000;
+  const expirationTimestamp = Math.floor(Date.now() / 1000) + 10000;
+  const siphonDelay = 100000;
+  const priceTrackingIdentifier = utf8ToHex("UMATEST");
+
+  beforeEach(async function() {
+    // Create and mint collateral token.
+    collateral = await MarginToken.new({ from: collateralOwner });
+    await collateral.addMember(1, collateralOwner, { from: collateralOwner });
+    await collateral.mint(sponsor, toWei("1000000"), { from: collateralOwner });
+    await collateral.mint(other, toWei("1000000"), { from: collateralOwner });
+
+    store = await Store.deployed();
+
+    // Create identifier whitelist and register the price tracking ticker with it.
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    await identifierWhitelist.addSupportedIdentifier(priceTrackingIdentifier, {
+      from: contractDeployer
+    });
+
+    // Create the instance of the PricelessPositionManager to test against.
+    // The contract expires 10k seconds in the future -> will not expire during this test case.
+    pricelessPositionManager = await PricelessPositionManager.new(
+      true, // _isTest
+      expirationTimestamp, // _expirationTimestamp
+      withdrawalLiveness, // _withdrawalLiveness
+      siphonDelay, // __siphonDelay
+      collateral.address, // _collateralAddress
+      Finder.address, // _finderAddress
+      priceTrackingIdentifier, // _priceFeedIdentifier
+      syntheticName, // _syntheticName
+      syntheticSymbol, // _syntheticSymbol
+      { from: contractDeployer }
+    );
+  });
+
+  it("Precision loss due to basic fees", async function() {
+    // Here, we choose a collateral amount that will produce one rounding error on the first run:
+    // - Collateral = 9,000,000 wei.
+    // - 0.0000004% fees per second * 1 second * 9,000,000 wei collateral = 3.5 wei fees, however this gets floored by `Store.computeFee()` to 3 wei fees.
```

Comment:
This should be 3.6 wei
![image](https://user-images.githubusercontent.com/12886084/76353251-0d7f4c00-6319-11ea-8043-61a1da9dd4af.png)
```suggestion
    // - 0.0000004% fees per second * 1 second * 9,000,000 wei collateral = 3.6 wei fees, however this gets floored by `Store.computeFee()` to 3 wei fees.
```

Reply:
ah yes